components/EmailValidator.vue

```diff
@@ -120,15 +122,30 @@ export default {
           }
         }
 
-        const validate = await this.$refs.email?.validate()
-        if (validate) {
-          this.$emit('update:valid', validate.valid)
+        const meta = this.$refs.form?.getMeta()
```

Comment:
I'm a bit worried by `form?`.  If that ref doesn't exist at this point, wouldn't we just quietly fail and never emit?

I don't think it should happen - but if it shouldn't happen, we don't need the `?`.

Reply:
Yeah, `?` is extra here. It confuses me when coming from TS, 'cause there a ref can always be undefined and you always need to provide either `?` or `!` when you 100% know this item is in DOM at given moment.